ios/engine/KMEI/KeymanEngine/Classes/KeymanWebViewController.swift

```diff
@@ -247,6 +247,31 @@ extension KeymanWebViewController {
     log.debug("Keyboard stub: \(stubString)")
     webView!.evaluateJavaScript("setKeymanLanguage(\(stubString));", completionHandler: nil)
   }
+    
+    func setLexicalModel(_ lexicalModel: InstallableLexicalModel) {
+        let stub: [String: Any] = [
+            "KI": "LexicalModel_\(lexicalModel.id)",
+            "KN": lexicalModel.name,
+            "KLC": lexicalModel.languageID,
+            "KL": lexicalModel.languageName,
+            "KF": storage.lexicalModelURL(for: lexicalModel).absoluteString
+        ]
+        
+        let data: Data
+        do {
+            data = try JSONSerialization.data(withJSONObject: stub, options: [])
+        } catch {
+            log.error("Failed to serialize lexical model stub: \(error)")
+            return
+        }
+        guard let stubString = String(data: data, encoding: .utf8) else {
+            log.error("Failed to create stub string")
+            return
+        }
+        
+        log.debug("LexicalModel stub: \(stubString)")
+        webView!.evaluateJavaScript("setKeymanLanguage(\(stubString));", completionHandler: nil)
```

Comment:
This isn't calling the correct API code point for lexical models.  I'd advise referring to #1699 and the changes in the following locations:

* keyboard.html - a JS-side helper for registering the stub with KMW.
* KMManager.java
  - (new) lines 653-669 - creates a proper 'lexical model stub',
  - (new) lines 675-680 - linking to that keyboard.html method for registering the stub.

It's mostly a matter of getting the structures and function names right.

Reply:
I've addressed these concerns pre-emptively in #1737, which will be based on this PR.